pkg/config/piped.go

```diff
@@ -181,24 +184,38 @@ func (s *PipedSpec) EnableDefaultKubernetesCloudProvider() {
 	s.CloudProviders = append(s.CloudProviders, defaultKubernetesPlatformProvider)
 }
 
-// HasCloudProvider checks whether the given provider is configured or not.
-func (s *PipedSpec) HasCloudProvider(name string, t model.ApplicationKind) bool {
-	requiredProviderType := t.CompatibleCloudProviderType()
-	for _, cp := range s.CloudProviders {
-		if cp.Name != name {
-			continue
+// HasPlatformProvider checks whether the given provider is configured or not.
+// Note: Currently check both list CloudProviders & PlatformProviders for configuration but we
+// will disable CloudProviders support next few versions.
+func (s *PipedSpec) HasPlatformProvider(name string, t model.ApplicationKind) bool {
+	requiredProviderType := t.CompatiblePlatformProviderType()
+	for _, cp := range s.PlatformProviders {
+		if cp.Name == name && cp.Type == requiredProviderType {
+			return true
 		}
-		if cp.Type != requiredProviderType {
-			continue
+	}
+	for _, pp := range s.CloudProviders {
+		if pp.Name == name && pp.Type == requiredProviderType {
+			return true
 		}
-		return true
 	}
 	return false
 }
 
-// FindCloudProvider finds and returns a Cloud Provider by name and type.
-func (s *PipedSpec) FindCloudProvider(name string, t model.ApplicationKind) (PipedPlatformProvider, bool) {
-	requiredProviderType := t.CompatibleCloudProviderType()
+// FindPlatformProvider finds and returns a Platform Provider by name and type.
+// Note: Currently check both list CloudProviders & PlatformProviders for configuration but we
+// will disable CloudProviders support next few versions.
+func (s *PipedSpec) FindPlatformProvider(name string, t model.ApplicationKind) (PipedPlatformProvider, bool) {
+	requiredProviderType := t.CompatiblePlatformProviderType()
+	for _, p := range s.PlatformProviders {
+		if p.Name != name {
+			continue
+		}
+		if p.Type != requiredProviderType {
+			continue
+		}
+		return p, true
+	}
 	for _, p := range s.CloudProviders {
 		if p.Name != name {
 			continue
```

Comment:
I feel that instead of keeping both fields as this.
Converting all configuration ways to use only a single `PlatformProviders` field could be easier to maintain.
I mean we can remove the `CloudProviders` field immediately from the Piped config and customizing the `Unmarhal` function to allow accepting both `cloudProviders` and `platformProviders` specification in yaml.

Reply:
And by that way, I guess that we can remove a lot of duplications that are being added in this PR.

Reply:
Nice idea, lets me try this 👍